wazuh/wazuh-dashboard-plugins
# Refactoring and redesign endpoints summary charts

## Description
This pull request modifies visualizations on Endpoints Summary changing the following:

- The Agents by State chart is refactored to be a decoupled and reusable component for other similar Endpoint Summary charts.
- The Agents by operating system graph is developed, reusing the refactored Agents by state component.
- The Agents by Group graph is developed, reusing the refactored Agents by State component.
- A card is developed showing the number of outdated agents.

Each visualization above has a user interaction that allows filters to be added to the table below as appropriate.

Additionally, the` /agents/stats/distinct` endpoint is updated in the imposter.
 
## Issues Resolved
#6258 

## Evidence
### Endpoints summary screen

### Breakpoint 1440px
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/17fda9ee-c538-4bea-a377-725122729f3e)

### Breakpoint 1024px
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/f1ff0a36-9ad1-40e6-9b1a-1e373ed65205)

### Breakpoint 768px
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/a6156f4b-51ea-4dbe-a6a2-dd8eed420064)


### Mobile
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/89d02e83-8d77-49c6-ae98-5170d7323538)



#### Endpoints summary screen with outdated agents popover opened
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/a41600ad-5e93-4d1a-ad0f-29b58b623a08)

#### Endpoints summary screen without outdated agents popover opened
![image](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/ce442732-428e-44ea-8919-b2422324adbb)


## Test

Legend:
⚫: none
🟢: pass
🟡: warning
🔴: fail
⚪: not applicable

### Steps to test:
- Go to Endpoints summary and test following:

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Check that the visualizations are rendered as shown in the evidence | :black_circle: | :black_circle: | :black_circle: |
| Check responsive version of visualizations rendering | :black_circle: | :black_circle: | :black_circle: |
| Check that the filter outdated agents button in the outdated agents component is disabled, using a real manager that does not have outdated agents | :black_circle: | :black_circle: | :black_circle: |
| Check that filters are applied when interacting with the Agents by status legend | :black_circle: | :black_circle: | :black_circle: |
| Check that filters are applied when interacting with the Agents by OS legend | :black_circle: | :black_circle: | :black_circle: |
| Check that filters are applied when interacting with the Agents by Group legend | :black_circle: | :black_circle: | :black_circle: |
| Check that filters are applied when interacting with the Filter outdated agents button | :black_circle: | :black_circle: | :black_circle: |
| Check the link to the "How to update agents" documentation in the Outdated popover | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: Check that the visualizations are rendered as shown in the evidence</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Check responsive version of visualizations rendering</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Check that the filter outdated agents button in the outdated agents component is disabled, using a real manager that does not have outdated agents</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Check that filters are applied when interacting with the Agents by status legend</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Check that filters are applied when interacting with the Agents by OS legend</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Check that filters are applied when interacting with the Filter outdated agents button</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Check the link to the "How to update agents" documentation in the Outdated popover</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

### Check List
